### PR TITLE
Botiboti day01

### PR DESCRIPTION
**Pull request by botiboti**
Well it's a short one.

### CODE REVIEW

File: Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+use std::io;
+
+fn main() {
+    println!("Enter age.");
+
+    let mut guess = String::new();
```
Comment (smorci) [inline Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs]: Nit: Why is the variable called guess if it represents an age? I think there are more suitable names to represent what the variable holds.

**Review by smorci — COMMENTED** [review]
Very elegant and concise solution, I like it.

**Review by smorci — APPROVED** [review]

File: Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs (hunk 1 above)
Comment (brainrake) [reply Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs]: copypasta :)

**Comment by brainrake** [comment]
This solution not match the requirements of the exercise. The readme says:

> Create a function that takes the age and return the age in days.

Please do that. And write some tests for it.

**Comment by brainrake** [comment]
Please don't mix PRs. Please remove stuff from this PR that is not related to Day 1 solution. Also, `use nix` is unnecessary since `use flake` is already in there.

**Review by brainrake — CHANGES_REQUESTED** [review]
Please follow the requirements: write a function that converts values. And add tests.

**Comment by botiboti** [comment]
That function is called main, yes.

File: Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,16 @@
+use std::io;
+
+fn main() {
+    println!("Enter age.");
+
+    let mut age = String::new();
+
+    io::stdin()
+        .read_line(&mut age)
+        .expect("Failed to read line.");
```
Comment (brainrake) [inline Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs:12]: nice error handling

File: Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+    println!("You entered: {age}");
+
+    let days: i32 = age.trim().parse().expect("Not an integer.");
```
Comment (brainrake) [inline Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs]: does this correctly handle inputs over 180 years?

File: Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs [hunk 4]
```diff
@@ -1,5 +1,10 @@
 use std::io;
 
+fn calc(str: String) -> i32 {
```
Comment (brainrake) [inline Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs]: Please separate the input parsing from the calculation. Input parsing is part of the interface between the program and the user, while application logic works on normalized data.

File: Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs [hunk 5]
```diff
@@ -1,5 +1,10 @@
...
+    let days: i32 = str.trim().parse().expect("Not an integer.");
```
Comment (brainrake) [inline Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs]: This can crash. It's a function that turns years into days, it should not be possible to crash it.

File: Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs (hunk 3 above)
Comment (botiboti) [reply Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs]: yes, haven't you tried it?

File: Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs (hunk 4 above)
Comment (botiboti) [reply Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs]: done

File: Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs (hunk 5 above)
Comment (botiboti) [reply Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs]: parsing separated, stays in main

File: Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+fn main() {
+    println!("Enter age.");
+
+    let mut age = String::new();
+
+    io::stdin()
+        .read_line(&mut age)
+        .expect("Failed to read line.");
+
+    let age: i32 = age.trim().parse().expect("Not an integer.");
```
Comment (brainrake) [inline Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs:16]: Does a negative age make sense? If not, let's not represent it.

File: Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs (hunk 3 above)
Comment (brainrake) [reply Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs]: right, i miscalculated. it blows up at two billion (days), so it couldn't handle earth for example

Comment (brainrake) [reply Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs]:
```
❯ cargo run
Enter age.
1000000000
You entered: 1000000000.
thread 'main' panicked at src/main.rs:4:5:
attempt to multiply with overflow
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Comment (brainrake) [reply Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs]: We should
1. check the input data so it is not too big
2. maybe use a larger type for the result

Comment (brainrake) [reply Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs]: perhaps use bigint for the whole thing, it can't be overflowed

File: Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs (hunk 4 above)
Comment (brainrake) [reply Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs]: :+1:

File: Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs (hunk 5 above)
Comment (brainrake) [reply Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs]: :+1:

**Comment by brainrake** [comment]
Please add some tests. I think unit tests are enough (testing the function itself) since the exercise is to write a function. `main` is there just to try it.

**Review by brainrake — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 7]
```diff
@@ -0,0 +1,3 @@
+target/
+.envrc
```
Comment (brainrake) [inline .gitignore]: i don't think .envrc should be ignored, we use it to set up direnv in the top level directory

File: Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs [hunk 8]
```diff
@@ -0,0 +1,19 @@
+use std::io;
+
+fn calc(age: i32) -> i32 {
```
Comment (brainrake) [inline Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs]: Does a negative age make sense? If not, let's not represent it.

File: Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+    age * 365
```
Comment (brainrake) [inline Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs]: This can overflow.
```
❯ cargo run
Enter age.
1000000000
You entered: 1000000000.
thread 'main' panicked at src/main.rs:4:5:
attempt to multiply with overflow
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
We have to make sure the types support al possible operations on them. Options:
- validate input
- make the result type bigger
- return a Maybe or Result
- use bigint

I lean towards the last one.

**Review by brainrake — CHANGES_REQUESTED** [review]

File: Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs (hunk 3 above)
Comment (brainrake) [reply Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs]: yeah my bad, miscalculated. it's one 100 million.

File: .direnv/flake-profile [hunk 10]
```diff
@@ -0,0 +1 @@
+flake-profile-1-link
```
Comment (brainrake) [inline .direnv/flake-profile]: irrelevant to the exercise, please remove. also .direnv will be removed from the project as it should be .gitignored.

File: .direnv/flake-profile-1-link [hunk 11]
```diff
@@ -0,0 +1 @@
+/nix/store/ljg8f5qb9bpl2432lbjbrpq1ncksxi2g-nix-shell-env
```
Comment (brainrake) [inline .direnv/flake-profile-1-link]: irrelevant to the exercise, please remove. also .direnv will be removed from the project as it should be .gitignored.

File: Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs (hunk 8 above)
Comment (brainrake) [inline Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs]: Please name it something descriptive.

File: Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+    io::stdin()
+        .read_line(&mut age)
+        .expect("Failed to read line.");
+
+    let age: BigUint = read_parse(age);
+    println!("You entered: {}.", age);
+    println!("That's roughly {} days.", calc_days(age));
+}
+
+#[test]
```
Comment (smorci) [inline Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs]: Move tests to different module and annotate them

Src: https://doc.rust-lang.org/book/ch11-03-test-organization.html#the-tests-module-and-cfgtest

File: Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs [hunk 13]
```diff
@@ -0,0 +1,63 @@
+use num_bigint::{BigUint, ToBigUint};
+use std::io;
+
+// convert input string to BigUint
+fn read_parse(inp: String) -> BigUint {
+    let age: BigUint = inp.trim().parse().expect("Not a positive integer.");
```
Comment (smorci) [inline Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs]: Misleading error message. If I enter "aa" I would get "Not a positive integer" which is true but misleading.

File: flake.nix
Comment (smorci) [inline flake.nix:1]: Formatting change.
Steps to take: align on formatter to use (nixfmt or nixpkgs-fmt) -> switch to that formatter -> include formatting changes in different PR

File: .direnv/flake-profile
Comment (smorci) [inline .direnv/flake-profile:1]: Remove .direnv and .gitignore it

**Review by smorci — DISMISSED** [review]

File: Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs (hunk 13 above)
Comment (botiboti) [reply Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs]: What else should it be then?

Comment (botiboti) [reply Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs]: I think that's the only sentence you need. An age is a positive integer.

File: .gitignore (hunk 7 above)
Comment (brainrake) [reply .gitignore:3]: .envrc should not be .gitignored

File: Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs (hunk 13 above)
Comment (brainrake) [reply Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs]: Looks ok. Not sure what a better error message would be.

File: Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs [hunk 14]
```diff
@@ -0,0 +1,68 @@
...
+// turn age in years to days
+fn calc_days(age: BigUint) -> BigUint {
+    let days_in_a_year: BigUint = 365_i32
+        .to_biguint()
+        .expect("Something went wrong with the conversion 365: i32 -> BigUint.");
+    age * days_in_a_year
+}
+
+fn main() {
+    println!("Enter age.");
```
Comment (brainrake) [inline Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs]:
```suggestion
    println!("Enter age:");
```

File: Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs [hunk 15]
```diff
@@ -0,0 +1,68 @@
...
+
+// convert input string to BigUint
+fn read_parse(inp: String) -> BigUint {
+    let age: BigUint = inp.trim().parse().expect("Not a positive integer.");
+    age
+}
+
+// turn age in years to days
+fn calc_days(age: BigUint) -> BigUint {
+    let days_in_a_year: BigUint = 365_i32
```
Comment (brainrake) [inline Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs]: I think you can just [multiply a u32 with a biguint](https://docs.rs/num-bigint/latest/num_bigint/struct.BigUint.html#impl-Mul%3C%26u32%3E-for-%26BigUint)
Also, define a [constant](https://doc.rust-lang.org/rust-by-example/custom_types/constants.html) rather than a let binding

File: Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs [hunk 16]
```diff
@@ -0,0 +1,68 @@
...
+fn main() {
+    println!("Enter age.");
+
+    let mut age = String::new();
+
+    io::stdin()
+        .read_line(&mut age)
+        .expect("Failed to read line.");
+
+    let age: BigUint = read_parse(age);
```
Comment (brainrake) [inline Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs]: Please avoid shadowing bindings, it gets confusing. Eg. rename the first `age` to `input`.

File: Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs [hunk 17]
```diff
@@ -0,0 +1,68 @@
...
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn an_ordinary_human_age() {
+        let result = read_parse(String::from("50"));
+        let expectation: BigUint = 50_i32
```
Comment (brainrake) [inline Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs]: You can use `.unwrap()` in tests.
You can also [use `BigUint::from(42_u32)`](https://docs.rs/num-bigint/latest/num_bigint/struct.BigUint.html#impl-From%3Cu32%3E-for-BigUint).

File: Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs [hunk 18]
```diff
@@ -0,0 +1,68 @@
+use num_bigint::{BigUint, ToBigUint};
+use std::io;
+
+// convert input string to BigUint
+fn read_parse(inp: String) -> BigUint {
```
Comment (brainrake) [inline Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs]: Don't need to pass a `String` object, which copies the object (but not the &str data) on the stack). A `&String` reference is enough. Actually, don't need a `&String` reference, which sometimes means allocating an extra object, since a `&str` is enough.
Rule of thumb: use `&str` if you can.
`String` is an object that wraps a heap-allocated &str. Use `String` if you can't use `&str`, eg. if a function returns a String and the returned object needs to own the &str inside.

File: Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs [hunk 19]
```diff
@@ -0,0 +1,68 @@
...
+    let days_in_a_year: BigUint = 365_i32
+        .to_biguint()
+        .expect("Something went wrong with the conversion 365: i32 -> BigUint.");
+    age * days_in_a_year
+}
+
+fn main() {
+    println!("Enter age.");
+
+    let mut age = String::new();
```
Comment (brainrake) [inline Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs]: This is shadowed by another variable. Please rename it to eg. `input`.

File: Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs [hunk 20]
```diff
@@ -0,0 +1,68 @@
+use num_bigint::{BigUint, ToBigUint};
+use std::io;
+
+// convert input string to BigUint
+fn read_parse(inp: String) -> BigUint {
+    let age: BigUint = inp.trim().parse().expect("Not a positive integer.");
```
Comment (brainrake) [inline Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs]: This function should return a `Result` rather than crash. Panicing is ok eg. for smaller applications, but it should be done at the outemost layer practical (here `main`).

**Review by brainrake — CHANGES_REQUESTED** [review]
Great progress. A few more changes to go.

File: Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs [hunk 21]
```diff
@@ -0,0 +1,68 @@
...
+    age
+}
+
+// turn age in years to days
+fn calc_days(age: BigUint) -> BigUint {
```
Comment (brainrake) [inline Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs]: This function can take the `age` input as a read only reference, so the object doesn't need to be copied on the stack.

**Comment by brainrake** [comment]
`cargo clippy` is also happy :+1:

**Comment by brainrake** [comment]
Please remove the changes to `.direnv` from the PR.

File: Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs (hunk 15 above)
Comment (botiboti) [reply Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs]: done

**Review by brainrake — CHANGES_REQUESTED** [review]

File: Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs (hunk 16 above)
Comment (botiboti) [reply Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs]: done

File: Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs (hunk 18 above)
Comment (botiboti) [reply Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs:5]: done

File: Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs (hunk 19 above)
Comment (botiboti) [reply Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs:21]: done

File: Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs (hunk 21 above)
Comment (botiboti) [reply Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs:11]: done

File: .gitignore (hunk 7 above)
Comment (brainrake) [reply .gitignore:3]: .envrc should NOT be .gitignored

File: Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs (hunk 21 above)
Comment (brainrake) [reply Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs]: I may have been too eager to optimize this one, not sure if it is required here. It makes some things harder (eg. i can't map this function over a `Result<BigUint, e>`. We'll figure this one out with more experience.

File: Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs [hunk 22]
```diff
@@ -0,0 +1,61 @@
...
+    }
+
+    #[test]
+    #[should_panic]
+    fn not_a_positive_integer() {
+        read_parse(&String::from("boo")).unwrap();
+    }
+
+    #[test]
+    fn calc_days_for_earth() {
```
Comment (brainrake) [inline Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs]: we need more than one test for the actual function

File: Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs [hunk 23]
```diff
@@ -0,0 +1,61 @@
...
+    #[test]
+    fn age_of_earth() {
+        let result = read_parse(&String::from("4500000000"));
+        assert_eq!(result, Ok(BigUint::from(4500000000_u64)));
+    }
+
+    #[test]
+    #[should_panic]
+    fn not_a_positive_integer() {
+        read_parse(&String::from("boo")).unwrap();
```
Comment (brainrake) [inline Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs]: `should_panic` here will catch other panics as well, which is not ideal for tests.
Either add a filter to `should_panic`, or use `assert_eq` with `Err`.

File: Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs [hunk 24]
```diff
@@ -0,0 +1,61 @@
+use num_bigint::{BigUint, ParseBigIntError};
+use std::io;
+
+const DAYS_IN_YEAR: u64 = 365;
+
+// convert input string to BigUint
+fn read_parse(inp: &str) -> Result<BigUint, ParseBigIntError> {
+    let age = match inp.trim().parse::<BigUint>() {
```
Comment (brainrake) [inline Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs]: this `match` is useless, just return the `Result`

**Review by brainrake — CHANGES_REQUESTED** [review]
almost there

File: Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs (hunk 22 above)
Comment (botiboti) [reply Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs]: I will add for ages: 0, 1 and 100

Comment (smorci) [reply Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs]: When asserting multiple things on the same action, I think it is generally a good idea to do that in one unit test that tests the function `calc_days`.

[Triple A](http://wiki.c2.com/?ArrangeActAssert)

Unify `calc_zero, calc_one, calc_hundred` into one test proving that `calc_days` works as expected.

**Review by brainrake — APPROVED** [review]

File: Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs (hunk 22 above)
Comment (brainrake) [reply Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs]: Yes, you can assert several things in one unit test "on the same action". But it does not apply here, since the "action" (i'd rather call it "result" or "value") is calling `calc_days` with a specific number, and calling it with another number is a different action.
Here's an example where we assert multiple properties on a prepared test value.
```
let result = calc_days(BigUint::from(1_u64));
assert!(result > 300);
assert(result < 500);
assert_eq!(result, BigUint::from(365_u64));
```
Triple A link is not really relevant, it suggest to separate the stages of a test, not to assert several properties on one. It actually warns of "Test methods that try to test too many different things at once".
In conclusion, I don't think we should put calls to `calc_days` with different arguments into a sigle unit test, this is fine as is.
Actually this discussion revealed a problem with the tests, which is testing too many things at once, see below.

File: Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs [hunk 25]
```diff
@@ -0,0 +1,82 @@
...
+
+    #[test]
+    #[should_panic(expected = "ParseBigIntError")]
+    fn not_a_positive_integer() {
+        read_parse(&String::from("boo")).unwrap();
+    }
+
+    #[test]
+    fn calc_days_for_earth() {
+        let result = calc_days(&read_parse(&String::from("4500000000")).unwrap());
```
Comment (brainrake) [inline Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs]:
```suggestion
        let result = calc_days(BigUint::from(4500000000_u64));
```
All the tests for `calc_days` also use `read_parse`, which may be testing too many things at once. We want to test the target function in isolation as much as possible. Also for the other tests.

**Review by brainrake — CHANGES_REQUESTED** [review]

File: Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs [hunk 26]
```diff
@@ -0,0 +1,82 @@
...
+const DAYS_IN_YEAR: u64 = 365;
+
+// convert input string to BigUint
+fn read_parse(inp: &str) -> Result<BigUint, ParseBigIntError> {
+    let age = inp.trim().parse::<BigUint>();
+    age
+}
+
+// turn age in years to days
+fn calc_days(age: &BigUint) -> BigUint {
```
Comment (brainrake) [inline Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs]: maybe this should be in a separate module and exported, pending slack discussion

File: Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs [hunk 27]
```diff
@@ -0,0 +1,82 @@
...
+    age * DAYS_IN_YEAR
+}
+
+fn main() {
```
Comment (brainrake) [inline Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs:5]: maybe this should be a cargo `example`, pending slack discussion

File: Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs (hunk 22 above)
Comment (smorci) [reply Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs]: @brainrake 
I get each of your points and agree that the Triple A link is not really proving my point. 

I however do not see why it would be a different action if the value of the function argument changes. To me it is just repetition. Also this way when looking at `cargo test` I do not know what action am I testing when I see `test tests::calc_zero` 

I want progress here so I am not going to block anyone, but I wanted to get my opinion through.

File: Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs (hunk 27 above)
Comment (brainrake) [reply Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs:5]: no, it's fine.

File: Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs (hunk 22 above)
Comment (brainrake) [reply Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs]: An action is something you do, like a calculation. The result can have different properties (like being less than or more than a given number, etc.). Calling a function with a different argument will produce a different result, which will have different properties.
Sometimes it makes sense to have several calls, such as in property-based testing (like QuickCheck), where the function is called with many random values and the result is checked for correctness properties. Another example is a macro like `#[test_value 1 => 365]`(I saw this somewhere, don't rememeber the name) that makes a test that calls the function, and several example inputs and outputs can be given. This is not one of those times.
@Smorci please resolve this conversation and approve the PR if you're more convinced than not.

**Review by brainrake — APPROVED** [review]
yeeeeaahh

**Review by smorci — APPROVED** [review]
